src/main/java/net/dv8tion/jda/api/entities/User.java

```diff
@@ -225,4 +228,171 @@ default String getEffectiveAvatarUrl()
+         * {@link net.dv8tion.jda.api.entities.User.UserFlag#UNKNOWN UserFlag.UNKNOWN} is returned.
+         * 
+         * @param  offset
+         *         The offset to match a {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlag} to.
+         *         
+         * @return {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlag} relating to the provided offset.
+         */
+        @Nonnull
+        public static UserFlag getFromOffset(int offset)
+        {
+            for (UserFlag flag : values())
+            {
+                if (flag.offset == offset)
+                    return flag;
+            }
+            return UNKNOWN;
+        }
+        
+        /**
+         * A set of all {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlags} that are specified by this raw int representation of
+         * flags.
+         * 
+         * @param  flags
+         *         The raw {@code int} representation if flags.
+         *         
+         * @return Possibly-empty EnumSet of {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlags}.
+         */
+        @Nonnull
+        public static EnumSet<UserFlag> getFlags(int flags)
+        {
+            final EnumSet<UserFlag> foundFlags = EnumSet.noneOf(UserFlag.class);
+            
+            if (flags == 0)
+                return foundFlags; //empty
+            
+            for (UserFlag flag : values())
+            {
+                if (flag != UNKNOWN && (flags & flag.raw) == flag.raw)
+                    foundFlags.add(flag);
+            }
+                    
+            return foundFlags;
+        }
+
+        /**
+         * This is effectively the opposite of {@link #getFlags(int)}, this takes 1 or more {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlags}
+         * and returns the raw offset {@code int} representation of the flags.
+         * 
+         * @param  flags
+         *         The array of flags of which to form into the raw int representation.
+         *         
+         * @return Unsigned int representing the provided flags.
+         */
+        public static int getRaw(@Nonnull UserFlag... flags){
+            int raw = 0;
+            for (UserFlag flag : flags)
+            {
+                if (flag != null && flag != UNKNOWN)
+                    raw |= flag.raw;
+            }
+            
+            return raw;
+        }
+
+        /**
+         * This is effectively the opposite of {@link #getFlags(int)}, this takes 1 or more {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlags}
+         * and returns the raw offset {@code int} representation of the flags.
+         * <br>Example: {@code getRaw(EnumSet.of(UserFlag.STAFF, UserFlag.HYPESQUAD))}
+         *
+         * @param  flags
+         *         The Collection of flags of which to form into the raw int representation.
+         *
+         * @return Unsigned int representing the provided flags.
```

Comment:
Java doesn't support unsigned ints. Call it `bitmask`

Reply:
So `Unsigned int` to `Bitmask`, or `Unsigned bitmask`?

Reply:
Just `bitmask` and yes best to refer to it as `bitmask` everywhere

Reply:
changed